File: lab4/Figures/Domain/Parallelepiped/Parallelepiped.cpp
#include "Parallelepiped.hpp"
#include "../../../../Utils/StringUtils.hpp"

double Parallelepiped::GetHeight() const
{
    return height;
}
double Parallelepiped::GetWidht() const
{
    return width;
}
double Parallelepiped::GetDepth() const
{
    return depth;
}
Parallelepiped::operator std::string() const
{
    return StringUtils::StringConcatenator() << density <<  ' ' << volume << ' ' << GetMass() << height << ' ' << width << ' ' << depth;
}

File: lab4/Bodies/lib/Figures/Parallelepiped/Parallelepiped.dart
import '../SolidBody/SolidBody.dart';

class Parallelepiped extends SolidBody
{
  double _height;
  double _depth;
  double _width;

  Parallelepiped(double density, double volume, double this._height, double this._depth, double this._width):
  super(density, volume) {}

    double GetHeight() => _height;
    double GetWidht() => _width;
    double GetDepth() => _depth;

    @override
  String toString() {
    return 'Parallelepiped';
  }
}
